File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.cpp
class Solution {
public:
    bool uniformArray(vector<int>& nums1) {
        int odd = 0 ;
        int mn = INT_MAX ;

        for(int i : nums1) {
            if(i % 2 != 0) {
                odd++ ;
            }

            mn = min(mn, i) ;
        }

        return mn % 2 != 0 || odd == 0 ;
    }
};

File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.java
class Solution {
    public boolean uniformArray(int[] nums1) {
        int oddCount = 0, minElement = Integer.MAX_VALUE ;

        for(int i : nums1) {
            if(i % 2 != 0) {
                oddCount++ ;
            }

            if(i < minElement) {
                minElement = i ;
            }
        }

        return minElement % 2 != 0 || oddCount == 0 ;
    }
}